internal/k8s/clients.go

```diff
@@ -84,3 +103,31 @@ func (c *Clients) ClientSet() *kubernetes.Clientset {
 func (c *Clients) DynamicClient() dynamic.Interface {
 	return c.dynamic
 }
+
+// APIResources holds a map of API, GroupVersionResources, that exist in the Kubernetes cluster
+type APIResources struct {
```

Comment:
I think that this types makes sense.

Looking at `DefaultResources()`, `ServiceAPIResources()`, etc, we could (in a followup PR) refactor those to return `APIResources`, then we have a consistent type for all this stuff.